x/rollup/keeper/msg_server.go

```diff
@@ -83,3 +86,70 @@ func (k *Keeper) InitiateWithdrawal(
 
 	return &types.MsgInitiateWithdrawalResponse{}, nil
 }
+
+func (k *Keeper) InitiateFeeWithdrawal(
+	goCtx context.Context,
+	_ *types.MsgInitiateFeeWithdrawal,
+) (*types.MsgInitiateFeeWithdrawalResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// TODO: make minWithdrawalAmount and l1recipientAddr configurable once a genesis state is added
+	const (
+		minWithdrawalAmount   = 100
+		l1recipientAddr       = "0x63d93aC6FA6B4021527e967ac3Eb29F2B3E52B96"
+		feeWithdrawalGasLimit = 400_000
+	)
+
+	feeCollectorAddr := k.accountkeeper.GetModuleAddress(authtypes.FeeCollectorName)
+	if feeCollectorAddr == nil {
+		ctx.Logger().Error("Failed to get fee collector address")
+		return nil, types.WrapError(types.ErrInitiateFeeWithdrawal, "failed to get fee collector address")
+	}
+
+	feeCollectorBalance := k.bankkeeper.GetBalance(ctx, feeCollectorAddr, types.WEI)
+	if feeCollectorBalance.Amount.LT(math.NewInt(minWithdrawalAmount)) {
+		ctx.Logger().Error("Fee collector balance is below the minimum withdrawal amount", "balance", feeCollectorBalance.String())
```

Comment:
Do we know what the purpose of having a minimum is?

Reply:
The [spec](https://specs.optimism.io/protocol/predeploys.html#sequencerfeevault) is kind of vague on the purpose of the minimum withdrawal amount:
> When enough fees accumulate in this account, they can be withdrawn to an immutable L1 address.

Since any L2 user can initiate a fee withdrawal, I'm guessing it's a safeguard to prevent a potential attack where an L2 user could spam fee withdrawals and then the sequencer would be paying more in gas on L1 than they would receive in fees.